README.md

```diff
@@ -6,3 +6,112 @@ AiiDA plugin that simplifies working with pseudo potentials.
+        pseudo = UpfData(stream)
+
+Note, the pseudopotential constructor expects a stream of bytes, so be sure to open the file handle with the `'rb'` mode.
+Alternatively, you can also use the `get_or_create` classmethod, which will first search the database if a pseudopotential of the exact same type and content already exists.
+If that is the case, that node is returned, otherwise a new instance is created:
+
+    from aiida import plugins
+
+    UpfData = plugins.DataFactory('pseudo.upf')
+
+    with open('path/to/pseudo.upf', 'rb') as stream:
+        pseudo = UpfData.get_or_create(stream)
+
+If a new node was created, it will be unstored.
+Note that if more than one pseudopotentials in the database are matched, a random one is selected and returned.
+
+
+### Families
+
+The pseudopotentials are organized in families, which are implemented as group plugins and the base class is `PseudoPotentialFamily`.
+A family class can in principle support many pseudopotential formats, however, a family instance can only contain pseudopotentials of a single format.
+For example the `PseudoPotentialFamily` can be created with any of the pseudopotential formats that are supported by this plugin, however, any instance can only have pseudopotentials of one single format.
+In contrast, the `SsspFamily` only supports the `UpfData` format.
+
+A pseudopotential family can be constructed manually, by first constructing the class instance and then adding pseudopotential data nodes to it:
+
+    from aiida import plugins
+
+    UpfData = plugins.DataFactory('pseudo.upf')
+    PseudoPotentialFamily = plugins.GroupFactory('pseudo')
+
+    pseudos = []
+
+    for filepath in ['Ga.upf', 'As.upf']:
+        with open(filepath, 'rb') as stream:
+            pseudo = UpfData(stream)
+            pseudos.append(pseudo)
+
+    family = PseudoPotentialFamily(label='pseudos/upf').store()
+    family.append(pseudos)
+
+Note that as with any `Group`, it has to be stored before nodes can be added.
+If you have a folder on disk that contains various pseudopotentials for different elements, there is an even easier way to create the family automatically:
+
+    from aiida import plugins
+
+    UpfData = plugins.DataFactory('pseudo.upf')
+    PseudoPotentialFamily = plugins.GroupFactory('pseudo')
+
+    family = PseudoPotentialFamily('path/to/pseudos', 'pseudos/upf', pseudo_type=UpfData)
+
+The plugin is not able to reliably deduce the format of the pseudopotentials contained in the folder, so one should indicated what data type to use with the `pseudo_type` argument.
+The exception is when the family class only supports a single pseudo type, such as for the `SsspFamily`, in which case that type will automatically be selected.
+Subclasses of supported pseudo types are also accepted.
+For example the base class `PseudoPotentialFamily` supports pseudopotentials of the `PseudoPotentialData` type, which, since it is the base class for all types, it will accept any pseudopotential type.
```

Comment:
For example, the base class `PseudoPotentialFamily` supports pseudopotentials of the `PseudoPotentialData` type. Because all more specific pseudopotential types are subclasses of `PseudoPotentialData`, the `PseudoPotentialFamily` class accepts all of them.